javasource/databaseconnector/impl/JdbcConnector.java

```diff
@@ -28,87 +30,145 @@
+
+    public JdbcConnector(final ILogNode logNode, final ObjectInstantiator objectInstantiator,
+                         final ConnectionManager connectionManager) {
+        this.logNode = logNode;
+        this.objectInstantiator = objectInstantiator;
+        this.connectionManager = connectionManager;
+    }
+
+    public JdbcConnector(final ILogNode logNode) {
+        this(logNode, new ObjectInstantiatorImpl(), ConnectionManagerSingleton.getInstance());
+    }
+
+    public Stream<IMendixObject> executeQuery(final String jdbcUrl, final String userName, final String password,
+                                              final IMetaObject metaObject, final String sql, final IContext context) throws SQLException {
+        return executeQuery(jdbcUrl, userName, password, metaObject, sql, new ArrayList<>()).map(CreateMendixObjectConverter(context, metaObject));
+    }
+
+    public Stream<IMendixObject> executeQuery(final String jdbcUrl, final String userName, final String password,
+                                              final IMetaObject metaObject, final IStringTemplate sql, final IContext context) throws SQLException {
+
+        String queryTemplate = sql.replacePlaceholders((placeholderString, index) -> "?");
+        List<ITemplateParameter> queryParameters = sql.getParameters();
+
+        return executeQuery(jdbcUrl, userName, password, metaObject, queryTemplate, queryParameters).map(CreateMendixObjectConverter(context, metaObject));
+    }
+
+    private Function<Object, Object> toSuitableValue(final PrimitiveType type) {
+        return v -> type == PrimitiveType.Binary ? new ByteArrayInputStream((byte[]) v) : v;
+    }
+
+    private Stream<Map<String, Optional<Object>>> executeQuery(final String jdbcUrl, final String userName,
+                                                               final String password, final IMetaObject metaObject, final String queryTemplate, final List<ITemplateParameter> queryParameters) throws SQLException {
+        logNode.trace(String.format("executeQuery: %s, %s, %s", jdbcUrl, userName, queryTemplate));
+
+
+        try (Connection connection = connectionManager.getConnection(jdbcUrl, userName, password);
+             PreparedStatement preparedStatement = connection.prepareStatement(queryTemplate)) {
+
+            addPreparedStatementParameters(queryParameters, preparedStatement);
+
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                ResultSetReader resultSetReader = new ResultSetReader(resultSet, metaObject);
+                return resultSetReader.readAll().stream();
+            }
+        }
+    }
+
+    private void addPreparedStatementParameters(List<ITemplateParameter> queryParameters, PreparedStatement preparedStatement) throws SQLException {
+        for (int i = 0; i < queryParameters.size(); i++) {
+            ITemplateParameter parameter = queryParameters.get(i);
+
+            switch (parameter.getParameterType()) {
```

Comment:
Shall we take care of the unhandled case here?

Reply:
Done.